### ISSUE TITLE
PageSelector does not sort pages correctly when limited

### ISSUE THREAD

**Issue by Toflar** [contao/core#8577]
When the page selector widget is limited by setting root nodes, the `sorting` is not considered.

**Comment by leofeyer** [comment]
What kind of root nodes did you set? Just the ID of the root page or all IDs of all pages?

**Comment by Toflar** [comment]
Changelanguage ;)

**Comment by leofeyer** [comment]
That does not really help …

**Comment by leofeyer** [comment]
https://github.com/terminal42/contao-changelanguage/blob/ad4269ece6918109cc4361bb4bf46a7fa9441ed7/library/Terminal42/ChangeLanguage/EventListener/DataContainer/PageInitializationListener.php#L117

**Comment by leofeyer** [comment]
Hm, I struggle to reproduce the issue.

```php
// system/config/dcaconfig.php
$GLOBALS['TL_DCA']['tl_user']['fields']['pagemounts']['eval']['rootNodes'] = array(49, 48, 1);
```

Even if I change the order of the IDs (e.g. `array(48, 1, 49)`), the sorting of the page picker tree is always correct.

**Comment by leofeyer** [comment]
The problem is that you are providing the IDs of all pages in the first level, which means you are setting different mounting points. If you instead provided just the ID of the root page, the problem would be solved:

```php
// Terminal42\ChangeLanguage\EventListener\DataContainer\PageInitializationListener:116
if (null !== $masterRoot) {
    $GLOBALS['TL_DCA']['tl_page']['fields']['languageMain']['eval']['rootNodes'] = $masterRoot->id;
}
```

**Comment by leofeyer** [comment]
Fixed in 97cddae5de4d532fc444f6ec375dd913a4d62d2b. You should still add `ORDER BY sorting` to the SQL query in the `PageInitializationListener` class on line 117.
